### PR TITLE
Update louvain distributed query

### PR DESCRIPTION
**Pull request by Boyu1997**
Summary of changes:
- Fix single vertex community issue.
- Add vertex following heuristic.

### CODE REVIEW

**Review by victor-gsl — DISMISSED** [review]

File: algorithms/schema-free/louvain-distributed/louvain_distributed_query.gsql [hunk 1]
```diff
@@ -1,150 +1,172 @@
...
-			s.@ccWeight += e.getAttr(wt_attr,"FLOAT")*1.0
-		POST-ACCUM 
-			s.@cc += s,
-			IF intermediateResult THEN s.@ccIntermediate += getvid(s) END
-	;
+    // initialize
+    Start = {v_type};
+    Start = SELECT s FROM Start:s -(e_type:e)- :t
+        ACCUM
+            @@m += e.getAttr(wt_attr, "INT")*0.5,
```
[the remark this answers is not in the document]
Comment (Boyu1997) [reply algorithms/schema-free/louvain-distributed/louvain_distributed_query.gsql]: Updated

File: algorithms/schema-free/louvain-distributed/louvain_distributed_query.gsql [hunk 2]
```diff
@@ -1,150 +1,172 @@
-CREATE QUERY louvain_distributed(SET<STRING> v_type, SET<STRING> e_type, STRING wt_attr, INT max_iter = 10, FLOAT tolerence = 0.0001,
-    BOOL intermediateResult = FALSE, BOOL print_accum = FALSE, STRING result_attr = "", STRING file_path = "") {
+CREATE DISTRIBUTED QUERY louvain_distributed(SET<STRING> v_type, SET<STRING> e_type, STRING wt_attr = "weight",
+    INT max_iter = 10, STRING result_attr = "cid", STRING file_path = "", BOOL print_info = FALSE) {
```
[the remark this answers is not in the document]
Comment (Boyu1997) [reply algorithms/schema-free/louvain-distributed/louvain_distributed_query.gsql]: Removed old Louvain

File: algorithms/schema-free/louvain-distributed/louvain_distributed_query.gsql [hunk 3]
```diff
@@ -1,150 +1,172 @@
...
+            IF s.@outdegree > 1 THEN s.@cc += s END;
+    IF print_info THEN
+        PRINT Start.size() AS AllVertexCount;
+    END;
+
+    // special @cc update in the first iteration
+    Start = SELECT t FROM Start:s -(e_type:e)- :t
+        WHERE s.@outdegree > 1 AND t.@outdegree > 1
+        ACCUM
+            t.@best_move += move(e.getAttr(wt_attr, "INT")*1.0 + @@m*t.@weight * (t.@weight - s.@weight), -s.@ccWeight, s.@cc.get(0))
```
[the remark this answers is not in the document]
Comment (Boyu1997) [reply algorithms/schema-free/louvain-distributed/louvain_distributed_query.gsql]: Updated

File: algorithms/schema-free/louvain-distributed/louvain_distributed_query.gsql [hunk 4]
```diff
@@ -1,150 +1,172 @@
...
+                            @@ccChange += 1
+                        END
+                    END
+                END;
+        IF print_info THEN
+            PRINT @@ccChange AS IterChangeCount;
+        END;
+    END;
+
+    // process node with outdegree = 1
```
[the remark this answers is not in the document]
Comment (Boyu1997) [reply algorithms/schema-free/louvain-distributed/louvain_distributed_query.gsql]: added comments

File: algorithms/schema-free/louvain-distributed/louvain_distributed_query.gsql [hunk 5]
```diff
@@ -1,150 +1,172 @@
...
+    Start = {v_type};
```
[the remark this answers is not in the document]
Comment (Boyu1997) [reply algorithms/schema-free/louvain-distributed/louvain_distributed_query.gsql]: added comments

File: algorithms/schema-free/louvain-distributed/louvain_distributed_query.gsql [hunk 6]
```diff
@@ -1,150 +1,172 @@
-CREATE QUERY louvain_distributed(SET<STRING> v_type, SET<STRING> e_type, STRING wt_attr, INT max_iter = 10, FLOAT tolerence = 0.0001,
-    BOOL intermediateResult = FALSE, BOOL print_accum = FALSE, STRING result_attr = "", STRING file_path = "") {
+CREATE DISTRIBUTED QUERY louvain_distributed(SET<STRING> v_type, SET<STRING> e_type, STRING wt_attr = "weight",
```
[the remark this answers is not in the document]
Comment (Boyu1997) [reply algorithms/schema-free/louvain-distributed/louvain_distributed_query.gsql]: updated to single-node mode as default; single-node and distributed will use the same algorithm
updated the query name to tg_louvain

File: algorithms/schema-free/louvain-distributed/louvain_distributed_query.gsql (hunk 2 above)
[the remark this answers is not in the document]
Comment (Boyu1997) [reply algorithms/schema-free/louvain-distributed/louvain_distributed_query.gsql]: Standardizing the information print to `INT verbosity` is a great idea, we should consider adding this to the styling guideline. @xchang2014 
Keeping `BOOL print_info` for this PR so we can merge it quickly.

**Review by YimingPan-Code — APPROVED** [review]

**Pull request merged**